Proposed Code of Conduct

brenns10 opened the pull request:
I didn't want the idea of a organizational code of conduct (as discussed in #21) to get forgotten.  Having one of these is pretty important, so I drafted something.  I feel that our group is very good, so this policy won't impact much day-to-day life - but having it in place is important in case problems do arise.

This is a pretty big deal, so I would like a lot of people to read this and comment.  This draft includes stuff from just me, and I would like it to reflect our organization as a whole.

---

edit from @timtadh
#### Process for Adoption

We are going to adopt this code of conduct by consensus between all participants on this draft. A broad call will go out on the main Hacker Society mailing list for input. When consensus is reached the code of conduct will be announced and adopted.
#### Target Date for Adoption

We hope to conclude the drafting and discussion process and adopt the code of conduct by March 16, 2016.
#### Process for Revision

Propose a change via a comment. Either a diff comment or a general comment on the PR. If some else approves then the change can become a commit. Note: if there disagreement about it majority consensus should be sought. If there is disagreement about a committed change make a comment on that commit and reference it from a comment in the main thread. If there is majority consensus that the commit should be reverted than a reverting commit should then be made. There should be no re-basing or force pushes on this branch!

mtbentley commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,71 @@
+-----------
+
+This policy applies at all Hacker Society events, such as weekly talks, open
+hacks, Link-State, HackCWRU, the mailing lists, and on our Slack team.  It
```
Also the github organization

brenns10 replied:
New commit incoming.

mtbentley commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,71 @@
+Enforcement
+-----------
+
+This policy applies at all Hacker Society events, such as weekly talks, open
```
s/events/events and spaces, both on- and off-line/

mtbentley commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,71 @@
+retain the right to take any actions to keep HacSoc a welcoming environment for
+all participants.  This includes warning the offender, or removing them from the
+event or communication streams they are abusing.  This also includes further
+action, such as pursuing University action, should it become necessary.
```
action...action
Sounds awkward

brenns10 commented:
Latest commit should address all those things.

mtbentley commented:
I like it :+1: 

I am happy to argue in favor of this code of conduct if needed.

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,71 @@
+physical appearance, body size, race, age, or religion.  As such, we do not
+tolerate harassment.  This includes, but is not limited to:
+
+- Rude or unwelcome comments
```
This should be clarified. I might find it unwelcome for someone to disparage my favorite editor, vim, I might even find it rude! However, it should be allowed. This is a silly example but one of the goals of an open society is to have freedom of thought and expression. This is necessarily in tension with freedom from rude or unwelcome comments. However, in the case of hacker society this is easily solved: comments attacking an individual on the grounds of: gender, gender identity, gender expression, attire, sexual orientation, disability, physical appearance, body size, body type, race, age, or religion are considered rude and unwelcome.

We should strive for politeness and civil discourse but we should also strive to make all welcome. Even those with unpopular ideas or thoughts. So when does an unpopular idea cross the line? To me this happens when it is an attack on an intrinsic characteristics of a person. Here is an example of an idea which attacks a _non-intrinsic_ characteristic: I hate how "Joe" is so bad at using email he always replies before quote text instead of after. The way "Joe" uses email is not intrinsic to Joe an may be violating other established community norms. However, saying: I hate how Joe always wears yellow shirts so ugly! That is saying something about an intrinsic to Joe their preference for yellow shirts. The former is allowed the latter would be viewed as unwelcome. In the same way saying: I love how Joe always uses vim, is fine but saying: I love how Joe always wears green it is so hot, is not fine.

This is a really tricky balance. In my views the goals are:
1. A welcoming environment
2. A non-sexualized environment
3. A free environment where people can explore ideas

Freedom is necessarily subservient to the other two. If not than the loudest and crassest have free reign. A community has goals and norms and an open society can have both and still be open. Open for who? All who want to engage with ideas not engage with attacking personal characteristics of other community members.

brenns10 replied:
@timtadh: I agree.  The original policy I was basing this on was a bit more specific - it actually spelled out the intrinsic characteristics you talk about.  Here is the original line (note it's something of a template):

> Verbal comments that reinforce social structures of domination [related to gender, gender identity and expression, sexual orientation, disability, physical appearance, body size, race, age, religion, [your specific concern here].

I made the line more general mostly because I was hoping it could apply to more undesirable conduct; in retrospect, I don't know what else it really would cover.  I was also unsure about the "social structures of domination" portion, since in my opinion, _any_ attack on someone based on their intrinsic characteristics is problematic, not just those that reinforce the "social structures of domination".  I might propose a simpler, but more specific wording:

> Verbal attacks or unwelcome comments about gender, gender identity and expression, sexual orientation, disability, physical appearance, body size, race, age, or religion

timtadh replied:
@brenns10 That looks better. I also don't know exactly what they meant by "structures of domination." That seems to be pulling in a philosophical theory that I am unfamiliar with. I would go with:

> Rude or unwelcome comments about gender, gender identity, gender expression, sexual orientation, disability, physical appearance, attire, race, age or religious expression.

or maybe better:

> Rude or unwelcome comments about a person's gender, gender identity, gender expression, sexual orientation, disability, physical appearance, attire, race, age or religious expression.

It might be valuable to have some examples and have some discussion of the tension in the document itself.

brenns10 commented:
Updated the point about "rude or unwelcome comments" and added a short paragraph exploring some of the tensions between providing an environment for open discussion, expression, and debate, and providing a welcoming environment.  @timtadh: I think this addresses our discussions well enough.

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,88 @@
+has some tension with the goal of providing a welcoming environment.  We expect
+that debate and constructive criticism are part of this open atmosphere.
+However, when these discussions devolve into attacks against a person and/or
+their intrinsic characteristics, we believe that this is unacceptable.
```
change to:

Necessarily, the goals of learning, community, and equality are in tension
with the goals individual empowerment, freedom of thought, and freedom of
expression. We strive for politeness and civil discourse but we also strive to
make _all_ welcome. Even those with unpopular ideas or thoughts.  However, in
order to foster our community certain limits on free expression must be
observed. We expect that debate and constructive criticism are part of this
open atmosphere. We expect that some members of the community may express
ideas which are not politically correct. We expect that some ideas expressed
by a minority may be deeply unpopular and distasteful to the majority.
However, if these ideas or discussions devolve into attacks against a person
and/or their intrinsic characteristics, we believe that this is unacceptable.

Here is an example of an idea which attacks a non-intrinsic characteristic: I
hate how Pat is so bad at using email he always replies before the quoted
text instead of after. The way Pat uses email is not intrinsic to Pat an may
be violating other established community norms. However, saying: I hate how
Pat always wears yellow shirts so ugly! That is saying something about an
intrinsic to Pat their preference for yellow shirts. The former is allowed the
latter would be viewed as unwelcome. In the same way saying: I love how Pat
always uses vim, is fine but saying: I love how Pat always wears green it is
so hot, is not fine.

brenns10 replied:
First paragraph ++, it adds a bit more detail that seems valuable.  The second paragraph I could take or leave - the examples don't seem (to me) that much more helpful.

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,88 @@
+
+We want Hacker Society to be a place where people can be comfortable regardless
+of gender, gender identity and expression, sexual orientation, disability,
+physical appearance, body size, race, age, or religion.  As such, we do not
```
To support Hacker Society's mission of: learning technology from each other,
building community, and promoting equality, Hacker Society must be a place
where people can be comfortable regardless of gender, gender identity and
expression, sexual orientation, disability, physical appearance, body size,
race, age, or religion. Therefore, we do not tolerate:

brenns10 replied:
Yeah, this is a bit more grounded.  I like it.

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,88 @@
+===============
+
+Hacker Society [#hacsoc]_ is, as Tim wrote in :doc:`hacsoc`, about an open
+society:
```
I would change this to restate the mission of Hacker Society.

# The Mission of Hacker Society

A time and place for people to learn about technology from each other, in order to build a community, to empower the individual, promote equality, and build an open society.

**Open Society**: An open society promotes freedom of thought and expression, personal responsibility, individualism, cooperation, and equality of all citizens.

brenns10 replied:
++

timtadh commented:
In 3d2a6d8 I accepted the changes based on @brenns10 comments.

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+Necessarily, the goals of learning, community, and equality are in tension
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
```
How about "We encourage civil discourse but we also strive to ..."

brenns10 replied:
added

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+Necessarily, the goals of learning, community, and equality are in tension
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
+make all welcome. Even those with unpopular ideas or thoughts. However, in
```
"Even those with unpopular..." - sentence fragment. Should be "make all welcome, including those with unpopular ideas"
"Ideas or thoughts" is redundant - stick to just "ideas"

brenns10 replied:
added

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
+make all welcome. Even those with unpopular ideas or thoughts. However, in
+order to foster our community certain limits on free expression must be
```
comma: "foster our community, certain limits..."

brenns10 replied:
added

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+Necessarily, the goals of learning, community, and equality are in tension
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
+make all welcome. Even those with unpopular ideas or thoughts. However, in
```
approved rephrasing.

pick one of "ideas, thoughts, views, opinions"

timtadh replied:
new version:

> , even those with unpopular ideas.

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
+make all welcome. Even those with unpopular ideas or thoughts. However, in
+order to foster our community certain limits on free expression must be
```
:+1:

timtadh replied:
:+1:

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+Necessarily, the goals of learning, community, and equality are in tension
+with the goals individual empowerment, freedom of thought, and freedom of
+expression. We strive for politeness and civil discourse but we also strive to
```
"We encourage polite, civil discourse, but we also strive to ..." ?

timtadh replied:
:+1: to @ajm188 suggestion.

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+ideas which are not politically correct. We expect that some ideas expressed
+by a minority may be deeply unpopular and distasteful to the majority.
+However, if these ideas or discussions devolve into attacks against a person
+and/or their intrinsic characteristics, we believe that this is unacceptable.
```
Reword to get rid of the word "believe" - this is a code of conduct and must be written with authority, not an expression of our beliefs. Also, you can't police ideas (1984 :) ), so focus on tangible things - people's expression of their ideas
"It is unacceptable if expression of these ideas devolve into attacks a person or his/her intrinsic characteristics"

On that note, may want to change "their" to "his or her" or "his/her" throughout
Well. that's what grammar says to do, but idk if Strunk & White of  _The Elements of Style_ fame knew about non-gender-specific pronouns :)

brenns10 replied:
> sweet!

It appears that the concensus is to leave this as is. A new issue can be opened for disagreement.

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+The Mission of Hacker Society
+=============================
+
+A time and place for people to learn about technology from each other, in order
```
start off talking about what organization we're talking about. Later on, you use HacSoc w/o defining the abbreviation. You can argue that we all know what HacSoc is, but I understand that this is supposed to be a more formal declaration.
"Hacker Society (HacSoc) at Case Western Reserve University is a time and place blah blah blah"

ajm188 replied:
:+1:

timtadh replied:
:+1: see my rewrite of the full paragraph in the next note down.

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+=============================
+
+A time and place for people to learn about technology from each other, in order
+to build a community, to empower the individual, promote equality, and build an
```
you're goin' cray with your infinitives here. 
"to build a community, to empower the individual, TO promote equality, and TO build an open society"

Also, you're using "build" twice. Replace one with a fun synonym like "foster" or "create!"

brenns10 replied:
used the rewrite

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+ideas which are not politically correct. We expect that some ideas expressed
+by a minority may be deeply unpopular and distasteful to the majority.
+However, if these ideas or discussions devolve into attacks against a person
+and/or their intrinsic characteristics, we believe that this is unacceptable.
```
I think we should keep the 'however', so "However, it is unacceptable if expression of these ideas ..."

timtadh replied:
Following the [Washington Post's decision](https://www.washingtonpost.com/opinions/the-post-drops-the-mike--and-the-hyphen-in-e-mail/2015/12/04/ccd6e33a-98fa-11e5-8917-653b65c809eb_story.html) we are using the singular their. It is quickly becoming the accepted gender neutral pronoun. His/her and his or her are both awkward constructions.

timtadh replied:
On the word "believe." To me this is a living document and statement of community norms. We are claiming this belief not just saying that this is unacceptable. This paragraph is meant to explain the reasoning and philosophy behind the code.

mrb113 replied:
sweet!

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+It is expected that members behave in a way that allows Hacker Society to
+achieve its mission.  Participants in Hacker Society events are expected to
+treat each other with dignity and respect for each other's perspectives,
```
you wouldn't say "I will treat you with respect for your perspectives", you'd say "I respect your perspective". 
Therefore... "respect each other's perspectives and treat each other with dignity" or something, idk

brenns10 replied:
updated

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+people of all skill levels are welcome, and should always be willing to help
+others learn.
+
+To support Hacker Society's mission of: learning technology from each other,
```
Split this up. 
"Hacker society's mssion is blah blah blah" (bullet points would be fun)
"To support this mission, Hacker Society must..."

brenns10 replied:
I ended up using

> To support its mission, Hacker Society must ...

because it felt a bit more natural. Same difference though.

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+It is expected that members behave in a way that allows Hacker Society to
+achieve its mission.  Participants in Hacker Society events are expected to
+treat each other with dignity and respect for each other's perspectives,
```
:+1:

timtadh replied:
:+1:

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+- Unwelcome sexual attention
+- Advocating for, or encouraging, any of the above behavior
+
+Necessarily, the goals of learning, community, and equality are in tension
```
yo this sentence is hard to understand. I had to read it a few times and it sounds very ivory tower-y once I figured it out. Also, I don't think that these goals are mutually exclusive as you have made it sound.
...can we just dump this sentence? I don't think it adds anything :)

brenns10 replied:
I very nearly added a line about a free code review from Michelle Bergeron, but left it unchanged :)

mrb113 replied:
thank God, someone would probably try taking me up on it

ajm188 replied:
@mrb113 please see: https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Aajm188+

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+Reporting
+---------
+
+If someone is making you feel unsafe or unwelcome at HacSoc, please report it as
```
Yo... so what if the person making you feel unsafe or unwelcome is an officer?

brenns10 replied:
Took the change, removed parenthetical.

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+As mentioned above, the policies of the `CWRU Handbook
+<https://students.case.edu/handbook/policy/>`_ are always applicable at HacSoc
+events and in HacSoc gathering spaces.  Other policies that we believe apply
```
do they apply here or do we believe they apply here?

brenns10 replied:
done

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+As mentioned above, the policies of the `CWRU Handbook
+<https://students.case.edu/handbook/policy/>`_ are always applicable at HacSoc
+events and in HacSoc gathering spaces.  Other policies that we believe apply
```
they apply

mrb113 replied:
:)

brenns10 replied:
I was the one who wrote "we believe they apply".  One reason because my writing style was dancing around a bit, which is my bad :/.  But the other reason I had to be hesitant was that these ACM and IEEE links aren't codes of conduct per se.  They are statements of ethics that are a ton broader than "how do we expect people to behave in HacSoc gatherings."  While most of the items in these documents are good things that I agree with, I think the purpose of our document is to lay out expectations for behavior at HacSoc events, not lay out a moral or ethical framework for our members.  Also, I haven't taken the time to fully read and digest these policies, and I'm not going to say "these certainly apply" without doing so.

timtadh replied:
So let's strike believe

> Other policies which apply are:

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+Reporting
+---------
+
+If someone is making you feel unsafe or unwelcome at HacSoc, please report it as
```
this is a really good question. who can they report to in this case? another officer? somewhere anonymous?

timtadh replied:
I suggest the department specifically Kim Yurchick as an alternate.

brenns10 replied:
This is a very good question.  Policies I've looked at before have mentioned anonymous reporting options.  We don't currently have one set up, but it was something I thought about bringing up.  Of course, the problem still remains that even with anonymous reporting, if the anonymous reports go to the officer list, the person you're reporting about still sees the message, and probably is more likely to connect the dots about who is reporting about them.  So I guess that leaves another officer.

If anyone has suggested strategies for this, I'd like to hear it.

mrb113 replied:
Another thing to think about is "What if I am an officer and someone is harassing me", especially if the other person is also an officer.
Probably straight to the department?

mrb113 replied:
I like the guidance of going to the department  At the end of the day, the officers are everyone's peers. Maybe they'll accidentally let something slip to another friend after a few $3 beers at the Scholar. Maybe the person who's been reported is their friend.
Lots of ickiness that could go on here.

Have we asked Kim what their guidance is? What situations do we have the authority to handle vs. what needs to go to the department/

mtbentley replied:
If an officer is being harassed by a non-officer, they should bring it up with the rest of the officers and the maintainer.  If it's between officers, it should be escalated to the next level.

mtbentley replied:
The next level could be the department or the Student Activities and Leadership Office (the office in charge of all student groups).

timtadh replied:
The university has anonymous reporting options. We should refer people to those. For conferences/hackCWRU it is a different matter (but they have their own CoC).

ajm188 replied:
I think, with Kim's permission, we should list her as an alternative.

timtadh replied:
Suggested change:

> If you don't feel comfortable reporting a problem to an officer (perhaps because an officer is the problem) consider reporting the problem directly to any of the faculty or staff of the EECS department.

wasianish replied:
Can we take that parenthetical and replace it with "for any reason"

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+offline.  This includes, but is not limited to:
+
+- Weekly talks
+- Open hacks
```
in emails, Open Hacks has both words capitalized

brenns10 replied:
updated

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+offline.  This includes, but is not limited to:
+
+- Weekly talks
+- Open hacks
```
:+1:

timtadh replied:
:+1:

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+- Weekly talks
+- Open hacks
+- Link-State
```
what is the process for changing this doc in the future? Say we rename Link-State to "The Thunderdome" and we start using the Pony Express to communicate instead of Slack? 

more general question that I don't care about the answer to, just want to make sure you've thought about an answer to the update process to this doc in the future.

brenns10 replied:
this sounds like a thing to discuss at the next GB as well

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+retain the right to take any actions to keep HacSoc a welcoming environment for
+all participants.  This includes warning the offender, or removing them from the
+event or communication streams they are abusing.  This also includes further
+measures, such as pursuing University action, should it become necessary.
```
who pursues university action? Am I supposed to if someone is harassing me at hacsoc or is HacSoc supposed to? Both?

brenns10 replied:
non-change here

mrb113 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+If someone is making you feel unsafe or unwelcome at HacSoc, please report it as
+soon as possible.  At events such as Link-State and HackCWRU, staff can be
+identified by their shirts.  At other events, you can talk to the officers or
+maintainer.  And you may always report harassing behavior to officers via email
```
maintainer is capitalized somewhere else

brenns10 replied:
updated

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+- Unwelcome sexual attention
+- Advocating for, or encouraging, any of the above behavior
+
+Necessarily, the goals of learning, community, and equality are in tension
```
This paragraph came out of this discussion: https://github.com/hacsoc/wiki/pull/22#discussion-diff-53735456 . I think it is important to say that yes there is conflict. It isn't mutually exclusive but there is conflict between those two goals. Freedom from being offended is absolutely in conflict with freedom of expression. We aren't going for the former but rather a very specific subset of the former.

brenns10 replied:
The way it is intended is not to say that there is a mutual exclusion, just a tension, like a sliding scale.  The more "freedom of expression" there is, the more potential for people saying things that are damaging to the community or harassing to people.  When there is lots of focus on maintaining a community without offensive things being said, there is less freedom of expression as a result.  So the idea is to find a balance where people can feel comfortable while also being able to freely express ideas.

Something like how the Bill of Rights says we have the right to free speech, but you still can't yell "fire" in a crowded movie theater.

At least, that's the way I thought this line was going.  It may not be how it's being received by readers though.

timtadh replied:
agree with @brenns10 explanation. My reasoning: we have had people in hacsoc who have been made rude comments but have also been valuable members of the community in other ways. College is a time when many people push boundaries on what is polite and acceptable. The larger hacker community also has many people who views that are sometimes significantly outside of the mainstream. We need a line in the sand to achieve our mission but we also need to recognize that their is a balance to strike. I will admit that it is a difficult balance to find.

mtbentley replied:
Although on the other hand, significant technical (or other) contributions should not excuse unacceptable behaviour.

brenns10 replied:
> Although on the other hand, significant technical (or other) contributions should not excuse unacceptable behaviour.

@matthewbentley: Totally agreed.  Two things to add to what @timtadh says about this:
1. In the cases Tim's talking about, I think people's behavior depends largely on who's receiving it.  If two people have a certain rapport where they are rude to each other in jest, and someone's comments are received that way, then I don't see a problem.  So long as others aren't being dragged in by wide-ranging offensive comments.
2. In my opinion, no amount of technical contribution excuses somebody from inappropriate, harassing behavior.  Someone who contributes technically but can't behave appropriately is clearly capable of behaving, but chooses not to.  They need to learn to behave appropriately before they can take part in the community.  I think this tradeoff is mostly important when people are discussing sensitive topics.  For example, discussions about gender, sexuality, etc., may be uncomfortable for some people, but so long as the discussion is being conducted in a respectful manner, they should be allowed.

timtadh replied:
> Although on the other hand, significant technical (or other) contributions should not excuse unacceptable behaviour.

That is not what is being said here. This code as written does not do that. What I said above does not do that. I am not doing that.

This attitude, "contribution is no excuse for bad behavior," is what this entire paragraph is about. People may have said something that a majority feels is politically incorrect outside of hacker society. Should that person be allowed to participate in hacker society? yes! As long as they don't violate the specific restrictions in place for hacker society.

It is when some one says or does something at hacker society that this policy comes into force. It is not about excluding people for behavior or statements outside of hacker society. For instance, we have had talks on http://urbit.org before by people who have interned there. Now it so happens that Curtis Yarvin the founder of urbit has an online persona unrelated to urbit or technology that some people find offensive. We should not bar Curtis from speaking on Urbit and on technology as long as he obeys the code of conduct and guidelines for speaking at hacker society.

This code of conduct is not about trying to ensure all hacker society participants morally or ethically upstanding. It is trying to ensure that:
1. People can learn about technology from each other
2. We can have a functioning respectful community
3. We can promote equality among the participants
4. We can promote freedom of thought and expression
5. We can foster an open society.

We can only do those things if:
1. We have a welcoming environment
2. We have a non-sexualized environment
3. A free environment where people can explore ideas

brenns10 replied:
> As long as they don't violate the specific restrictions in place for hacker society.

I think what @matthewbentley and I are talking about is that, if you're harassing someone, it doesn't really matter whether you make good technical contributions - you're harassing and need to stop.

What we judge to be politically correct is a bit more difficult, and certainly your personal beliefs and things you've said elsewhere don't have much bearing on HacSoc, so long as you comport yourself respectfully to others.

timtadh replied:
> My reasoning: we have had people in hacsoc who have been made rude comments but have also been valuable members of the community in other ways. College is a time when many people push boundaries on what is polite and acceptable.

I should note: the people and comments I am thinking of would not have violated the current code as written. They were rude but they were not attacking a persons intrinsic characteristics. As I have said there is a line in the sand. We should be clear where it is so that these people can also participate. The world would be the poorer without them.

mtbentley replied:
I think I agree, and I apologize if I misunderstood or am misunderstanding what you're saying.

As to what @timtadh said about behaviour and statements outside of HacSoc: while we cannot enforce a code of conduct outside of HacSoc venues, actions outside of HacSoc can still be relevant.  For example, if person A harasses person B outside of any HacSoc related venues, the presence of person B may make person A feel unwelcome at our events.  We would have to decide what actions, if any, this code of conduct would permit.

brenns10 replied:
The way I was interpreting this was a bit different than the way Tim was, as our separate comments reflect.  The line between having/stating an unpopular opinion, and attacking a person's intrinsic characteristics is pretty hard to define.

For instance, a person who is blatantly sexist, and believes women can't code, has an extremely unpopular opinion, (one which I find very difficult to accept or justify).  I suppose they could be allowed to participate, but the moment they voice that opinion, they are attacking the intrinsic characteristics of all women in our group, and that's not okay to me, nor is it acceptable by this code of conduct.

Tim, maybe you could help me understand what unpopular opinions you're talking about, where expression of them doesn't involve attacking intrinsic characteristics of our members.

mtbentley replied:
> I suppose they could be allowed to participate, but the moment they voice that opinion, ... that's not okay to me, nor is it acceptable by this code of conduct.

I would tend to agree with this.  In reality it is also worth considering the possible PR implications of hosting a talk by somebody with such views.  But that should not necessarily be covered by this CoC.

brenns10 replied:
Sounds like we're oscillating around agreement around this point.  It seems like the fact that we're having such a protracted discussion is evidence enough that there is a trade-off to be made here.  So I think we might as well leave this stuff as it is, and allow issues to be considered in the light of this trade-off when (if) issues do occur.

timtadh replied:
> For instance, a person who is blatantly sexist, and believes women can't code, has an extremely unpopular opinion, (one which I find very difficult to accept or justify). I suppose they could be allowed to participate, but the moment they voice that opinion, they are attacking the intrinsic characteristics of all women in our group, and that's not okay to me, nor is it acceptable by this code of conduct.

Agreed. That situation is one this CoC is meant to specifically address (and I think it does). If they do voice that opinion they would need to leave and not be invited back.

> I would tend to agree with this. In reality it is also worth considering the possible PR implications of hosting a talk by somebody with such views. But that should not necessarily be covered by this CoC.

Most of the time we wouldn't know beforehand. In the case that we did know about it before hand the maintainer would need to address the issue head on during the setup of the talk and also consider whether given the other proposals the topic is worth hearing. But in the case where some one is already scheduled for a talk and a potential issue comes to light what is the right course of action? The talk should still occur but the maintainer should before hand ask for a second confirmation of the CoC highlighting their concerns.

timtadh replied:
I would concur on leaving it unchanged unless someone has a clarifying suggestion. @matthewbentley and @mrb113 do you agree?

mrb113 replied:
> For instance, a person who is blatantly sexist, and believes women can't code, has an extremely unpopular opinion, (one which I find very difficult to accept or justify). 

Anyone who believes women cannot code is subject to one free code review from Michelle Bergeron. 

I agree that leaving it unchanged is probably fine; @timtadh

timtadh replied:
@mrb113 :heavy_check_mark: We will have you give them a code review!

Ok. Lets ship this non-change. :shipit:

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+people of all skill levels are welcome, and should always be willing to help
+others learn.
+
+To support Hacker Society's mission of: learning technology from each other,
```
I am restating the specific part of the mission which generates this restriction on expression.

brenns10 replied:
Since there's already a nice, bold section of "The Mission of Hacker Society" directly above this, I don't think we need to reiterate the mission all over again.  We can probably just say, "To support Hacker Society's mission, it must ..." and be fine.

timtadh replied:
Ok let's take @brenns10 suggestion.

> To support Hacker Society's mission, it must ...

:+1:

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+=============================
+
+A time and place for people to learn about technology from each other, in order
+to build a community, to empower the individual, promote equality, and build an
```
Rolling these previous to comments together full rewrite is:

> Hacker Society (HacSoc) at Case Western Reserve University is a time and place for people to learn about technology from each other, in order to build a community, to empower the individual, promote equality, and foster an open society.

ajm188 replied:
Looks better, but there's still an issue with the infinitives. See the all caps sections of @mrb113's comment above

timtadh replied:
> Hacker Society (HacSoc) at Case Western Reserve University is a time and place for people to learn about technology from each other, in order to build a community, empower the individual, promote equality, and foster an open society.

timtadh commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+retain the right to take any actions to keep HacSoc a welcoming environment for
+all participants.  This includes warning the offender, or removing them from the
+event or communication streams they are abusing.  This also includes further
+measures, such as pursuing University action, should it become necessary.
```
I didn't draft this section but from the context it seems that the leadership who asked the person to stop doing something is pursuing university action.

brenns10 replied:
I did write this.  I wanted to include University action as an explicit option for resolving disputes when necessary.  I wasn't trying to come up with a rule for who take this action, just explicitly state that University repercussions could happen for poor behavior at HacSoc.

It seems like regardless of _who_ appeals to the university, the person who is being harassed should be able to say whether or not they would like to pursue university action.

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+
+- Weekly talks
+- Open hacks
+- Link-State
```
For sake of documentation, see #23 if you haven't already.

zwh2 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+- HackCWRU
+- Mailing lists (hackers-discuss, cwru-hackers)
+- GitHub organization
+- Slack team
```
Do we want to claim IRC?

brenns10 replied:
non-change here

ajm188 commented on organizational/conduct.rst:
```diff
@@ -0,0 +1,101 @@
+- HackCWRU
+- Mailing lists (hackers-discuss, cwru-hackers)
+- GitHub organization
+- Slack team
```
As long as the current maintainers consent, I don't see why not.

yetanothername replied:
As the dude in charge of the IRC server right now: I do not consent to this policy being applied to IRC because there is already a notice that the server is under the CWRU Acceptable Use Policy.

timtadh replied:
I think that is fine.

zwh2 commented:
I will reach out to Andy. But this bring up a question if we want to extend this to other server stuff... Like could I host a website that violates?

mtbentley commented:
We have to decide, but I would lean towards no.  Right now I have arbitrary power to decide what does and does not go on the server, and I see no problem with declining our services toward individuals or groups who would violate this CoC.

zwh2 commented:
So should we put the servers in here some how. Someone will have to do the wording on that.

timtadh commented:
@zwh2 and @matthewbentley I think the servers should have a separate policy. There are a lot of other things we don't want running on the servers besides things that violate this CoC -- bitcoin mining, drive by malware sites, etc... So the server needs a separate policy.

zwh2 commented:
I would agree that we want additional restrictions on the server but I don't see why we shouldn't have this apply.

ajm188 commented:
Well, if we have additional restrictions on the server (which I think we should), then the server CoC would read something like:

```
Hacker Society CoC is in effect.
Additional Restriction #1
Additional Restriction #2
...
```

So there would be no need to do anything further in this PR.

timtadh commented:
@ajm188 agreed.

mtbentley commented:
I'm not sure how to structure the server(s) part of this.  On one hand, we should require externally facing sites to follow the CoC (we don't want to host anything that promotes harassment, racism, sexism, etc), but on the other hand it might be a bit heavy handed to require services such as IRC to adopt it.  Although if IRC markets itself as an official HacSoc service, then it should follow the CoC.

And obviously anything on the server would have to follow CWRU acceptable network usage policies, the law, and we would want a clause allowing us to turn down services for no stated reason if we want to.

mtbentley commented:
Also, let's move server discussion to #24

thomasrussellmurphy commented:
What's the status on this? The initial goal of March has flown through finals/graduation/summer.

Adding references to the resources from our relevant professional organizations would be a useful thing to have, as there is a significant community overlap at CWRU:
- http://www.acm.org/about-acm/code-of-ethics
- http://www.ieee.org/about/corporate/governance/p7-8.html
- http://www.ieee.org/about/ieee_code_of_conduct.pdf

brenns10 commented:
As a preliminary warning to the participants of this discussion - I will be going through the remaining commit comments, taking the agreed upon resolution, applying that, and pushing a new commit. This pull request is a year old and needs to go. **Therefore once that is done I will be merging what we have.**

I don't mean to close all discussion - anything you feel isn't completely settled, you can raise in new issues or pull requests. Their smaller scope should help facilitate a more productive conversation. This PR just reached a tipping point of size that exhausted everyone!

brenns10 commented:
Okay, so the updates according to the open comments are done. While I don't think that merging a pull request == adopting a policy, I suppose that I should not unilaterally merge something either.

I think the process for "deploying" a code of conduct ought to look like this:
1. I can add a bold "draft" notice at the top of the page, and then merge this in.
2. Push out a link to the mailing list, inviting people to discuss it at the next GB. Electronic discussion is nice too, but it could get unwieldy.
3. At the next GB, a vote would probably be in order. Especially since there is new exec and who knows how much of this they've seen.
4. The next issue to address would be how to update this - #23

brenns10 commented:
MERGING IT!!!
